src/plugins/data/public/ui/query_editor/query_editor_top_row.tsx

```diff
@@ -225,7 +225,17 @@ export default function QueryEditorTopRow(props: QueryEditorTopRowProps) {
   }
 
   function shouldRenderDatePicker(): boolean {
-    return Boolean(props.showDatePicker ?? true) ?? (props.showAutoRefreshOnly && true);
+    return (
+      Boolean(props.showDatePicker ?? true) &&
+      !(
+        queryLanguage &&
+        data.query.queryString.getLanguageService().getLanguage(queryLanguage)?.hideDatePicker
+      ) &&
+      (props.query?.dataset
+        ? data.query.queryString.getDatasetService().getType(props.query.dataset.type)?.meta
+            ?.supportsTimeFilter !== false
```

Comment:
```suggestion
            ?.supportsTimeFilter
```

i think?

Reply:
We want to let `undefined` values pass. Since, if `supportsTimeFilter` is `undefined`, then `.supportsTimeFilter` would be false. However, since `supportsTimeFilter` is an optional parameter, we want the date picker to still show when it is `undefined`  (it's currently only defined in the `s3_type`). This checks if the `supportsTimeFilter` flag is explicitly set to false, which is the only case where the date picker should be hidden.